BUILD.md

```diff
@@ -1,29 +1,54 @@
+# Requirements
+
+## Windows
+
+In order to build OmniSharp, the [.NET 4.6 targeting pack](http://go.microsoft.com/fwlink/?LinkId=528261) must be installed if it isn't already.
+
+## macOS
+
+**Mono 4.8.0** or greater is required. You can install this using the latest [.pkg](http://www.mono-project.com/download/#download-mac) or install it view [Homebrew](https://brew.sh/):
+
+```
+brew update
+brew install mono
+```
+
+## Linux
+
+Because OmniSharp uses the .NET Core SDK as part of the build, not all Linux distros are supported. A good rule of thumb is to check the list [here](https://www.microsoft.com/net/download/linux) to see if your particular distro is supported.
+
+**Mono 4.8.0** or greater is required. Each distro or derivative has it's own set of instructions for installing Mono which you can find [here](http://www.mono-project.com/download/#download-lin).
+
+In addition, the `msbuild` package must be installed. On Debian, Ubuntu, and derivatives, this can be achieved by first adding the Mono Project GPG signing key and package repository using the instructions [here](http://www.mono-project.com/docs/getting-started/install/linux/#debian-ubuntu-and-derivatives). Then, install msbuild via apt-get.
+
+```
+sudo apt-get install msbuild
+```
+
 # Usage
 
 Run `build.(ps1|sh)` with the desired set of arguments (see below for options).
-The build script itself is `build.cake`, written in C# using the Cake build automation system.
+The build script itself is `build.cake`, written in C# using the [Cake build automation system](http://cakebuild.net/).
 All build related activites should be encapsulated in this file for cross-platform access.
 
 # Arguments
 
-## Primary
-
-  `-target=TargetName`: The name of the build task/target to execute (see below for listing and details).
+  `-target TargetName`: The name of the build task/target to execute (see below for listing and details).
     Defaults to `Default`.
 
-  `-configuration=(Release|Debug)`: The configuration to build.
+  `-configuration (Release|Debug)`: The configuration to build.
     Defaults to `Release`.
 
-## Extra
-
-  `-test-configuration=(Release|Debug)`: The configuration to use for the unit tests.
+  `-test-configuration (Release|Debug)`: The configuration to use for the unit tests.
     Defaults to `Debug`.
-  
-  `-install-path=Path`: Path used for the **Install** target.
+
+  `-install-path Path`: Path used for the **Install** target.
     Defaults to `(%USERPROFILE%|$HOME)/.omnisharp/local`
-  
+
   `-archive`: Enable the generation of publishable archives after a build.
 
+Note: On macOS/Linux, be sure to pass the arguments above with double slashes! (e.g. `--target TargetName`).
```

Comment:
I've actually modified my cake scripts to have the first argument be the traget, much like grunt / gulp.

I don't have a shell version yet however.